Included the new versions of apriltag files

pmordel opened the pull request:
Hi Fabrizio, I upgraded the files from the apriltag-2016-12-01 version. It works for me but I had to modify the apriltag.h file in order to keep the '#include homography.h' wich has disappeared in the most recent version of the file. We need it to compile with catkin the 'node.cpp' since we use a homography function at line 522. I don't know why they removed this include from the apriltag.h

pmordel committed:
Added Makefile and setting proper parameters to CMakelists. So far I linked the apriltag_ros library to conmpile so I did not add it directly but it will have to be done.

pmordel committed:
Set path to the library on my station

pmordel committed:
Merge branch 'master' of https://github.com/pmordel/apriltag2

Conflicts:
	Makefile

pmordel committed:
Added missing dependencies to executable in CMakelists.txt

pmordel committed:
Commented all unnecessary lines in CMakelist. It now compiles only the 'apriltag2_exe' executable

pmordel committed:
Moved all unnecessary files in an archive and properly commented CMakeLists.txt

pmordel committed:
Update apriltag files from the apriltag-2016-12-01 release

pmordel committed:
added nez versions files. It does compile now.

pmordel committed:
Remove the line to install the library inside package

pmordel committed:
add README

pmordel committed:
Added #include "common/homography.h" in the new apriltag.h since we need it to compile with catkin "node.cpp" (issue with line 522)

pmordel committed:
Merge branch 'master' into master

fabrizioschiano commented:
Thanks Pol, I will check this as soon as possible and merge the pull request.

pmordel committed:
Added the following in order that all ros-packages would be found
include_directories(
    ${catkin_INCLUDE_DIRS}
)

pmordel commented:
I set up an all new architecture for the package in my last commit so you souldn't have any modification to do in CMakelist nor Makefile in the future, everything sould be good to work with now. Here is the description of my last commit : 

New architecture : Everything is now in the 'apriltag2_example' folder I modified the CMakeLists and Makefile in order that there is no need to modify them anymore (no path to change). In order to build the library, just go in the apriltag-2016-12-01 directory, create a build repository, go there and run 'ccmake ../../' and type 'c', 'e', 'c', 'e', 'g'. Then go back to the previous repository and run 'make' then 'sudo make install'.

I also modified the 'string_util.h' file :
-- #include "common/zarray.h"
++ #include "zarray.h"

pmordel commented:
next step would be to bring what's in the apriltag2_example at the root of the package. I'll do it as soon as you can check my previous commits so it works for you.

pmordel committed:
New architecture : Everything is now in the 'apriltag2_example' folder. I modified the CMakeLists and Makefile in order that there is no need to modify them anymore (no path to change). In order to build the library, just go in the apriltag-2016-12-01 directory, create a build repository, go there and run 'ccmake ../../' and type 'c', 'e', 'c', 'e', 'g'. Then go back to the previous repository and run 'make' then 'sudo make install'.

I also modified the 'string_util.h' file :
-- #include "common/zarray.h"
++ #include "zarray.h"
